### PR TITLE
fix: refactor signup hud into plugin

### PR DESCRIPTION
**Pull request by AjimenezDCL**
## How to test the changes?
If you login in a new session with `ENABLE_SEAMLESS_LOGIN` the avatar editor shouldnt appear. Additionally, check that without the feature flag the behaviour is the same as production.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### <samp>🤖 Generated by Copilot at 86749db</samp>

This pull request migrates the signup HUD from a HUD element to a plugin, which is a system that allows modular and independent features to be added to the project. The pull request removes the signup HUD logic from the `HUDController`, `HUDFactory`, and `HUDBridge` scripts, and creates a new plugin folder with the `SignupHUDPlugin` class and its related files. The pull request also updates the `SignupHUDController` script to improve its compatibility and functionality with the plugin system, and adds the signup HUD plugin as a reference and a dependency to the plugin system assembly.

### CODE REVIEW

File: unity-renderer/Assets/DCLPlugins/SignupHUDPlugin/SignupHUDPlugin.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        private const string SIGNUP_HUD = "SignupHUD";
+
+        private SignupHUDController controller;
+
+        public SignupHUDPlugin()
+        {
+            Initialize().Forget();
+        }
+
+        private async UniTaskVoid Initialize()
```
Comment (mikhail-dcl) [inline unity-renderer/Assets/DCLPlugins/SignupHUDPlugin/SignupHUDPlugin.cs:19]: I have a feeling somewhere here should be a `CancellationToken`, similar to `InitializeAsync` from `IService`?

**Review by mikhail-dcl — COMMENTED** [review]
I have a suggestion:

File: unity-renderer/Assets/DCLPlugins/SignupHUDPlugin/SignupHUDPlugin.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
+﻿using Cysharp.Threading.Tasks;
+using DCL;
+using DCL.Providers;
+using SignupHUD;
+
+namespace DCLPlugins.SignupHUDPlugin
+{
+    public class SignupHUDPlugin : IPlugin
```
Comment (lorux0) [inline unity-renderer/Assets/DCLPlugins/SignupHUDPlugin/SignupHUDPlugin.cs:8]: thank you =)

File: unity-renderer/Assets/DCLPlugins/SignupHUDPlugin/SignupHUDPlugin.cs (hunk 1 above)
Comment (lorux0) [reply unity-renderer/Assets/DCLPlugins/SignupHUDPlugin/SignupHUDPlugin.cs:19]: +1, but perhaps it out of the scope of this pr

**Review by lorux0 — APPROVED** [review]

**Review by kwsskulinski — APPROVED** [review]
As described, by using the ENABLE_SEAMLESS_LOGIN flag the avatar editor does not appear.

**Review by mikhail-dcl — APPROVED** [review]

**Pull request merged**